Cannot build programs with `-gc=none/leaking` on raspberry pi pico

Latest dev branch. Cannot build a simple program on pico nor pico2
```
tinygo build -target=pico2 -gc=none  .
# runtime
/home/pato/Documents/src/tg/tinygo/src/runtime/runtime_rp2.go:235:2: undefined: gcScanState
/home/pato/Documents/src/tg/tinygo/src/runtime/runtime_rp2.go:245:2: undefined: scanCurrentStack
/home/pato/Documents/src/tg/tinygo/src/runtime/runtime_rp2.go:252:2: undefined: gcScanState
/home/pato/Documents/src/tg/tinygo/src/runtime/runtime_rp2.go:262:2: undefined: gcScanState
```

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -778,6 +778,8 @@ endif
 	@$(MD5SUM) test.hex
 	$(TINYGO) build -size short -o test.hex -target=pico                examples/blinky1
 	@$(MD5SUM) test.hex
+	$(TINYGO) build -size short -o test.hex -target=pico -gc=leaking    examples/blinky1
+	@$(MD5SUM) test.hex
 	$(TINYGO) build -size short -o test.hex -target=nano-33-ble         examples/blinky1
 	@$(MD5SUM) test.hex
 	$(TINYGO) build -size short -o test.hex -target=nano-rp2040         examples/blinky1

--- a/src/machine/machine_rp2040_usb.go
+++ b/src/machine/machine_rp2040_usb.go
@@ -123,14 +123,12 @@ func handleUSBIRQ(intr interrupt.Interrupt) {
 	}
 }
 
-var empty [0]byte
-
 func handleUSBSetAddress(setup usb.Setup) bool {
 	// Using 570μs timeout which is exactly the same as SAMD21.
 	const ackTimeout = 570
 
 	rp.USBCTRL_REGS.SIE_STATUS.Set(rp.USBCTRL_REGS_SIE_STATUS_ACK_REC)
-	sendUSBPacket(0, empty[:])
+	sendUSBPacket(0, nil)
 
 	// Wait for transfer to complete with a timeout.
 	t := timer.timeElapsed()

--- a/src/machine/machine_rp2040_usb.go
+++ b/src/machine/machine_rp2040_usb.go
@@ -123,12 +123,14 @@ func handleUSBIRQ(intr interrupt.Interrupt) {
 	}
 }
 
+var empty [0]byte
+
 func handleUSBSetAddress(setup usb.Setup) bool {
 	// Using 570μs timeout which is exactly the same as SAMD21.
 	const ackTimeout = 570
 
 	rp.USBCTRL_REGS.SIE_STATUS.Set(rp.USBCTRL_REGS_SIE_STATUS_ACK_REC)
-	sendUSBPacket(0, []byte{})
+	sendUSBPacket(0, empty[:])
 
 	// Wait for transfer to complete with a timeout.
 	t := timer.timeElapsed()

--- a/src/machine/machine_rp2_usb.go
+++ b/src/machine/machine_rp2_usb.go
@@ -141,7 +141,7 @@ func setEPDataPID(ep uint32, dataOne bool) {
 }
 
 func SendZlp() {
-	sendUSBPacket(0, []byte{})
+	sendUSBPacket(0, empty[:])
 }
 
 func sendViaEPIn(ep uint32, data []byte, count int) {

--- a/src/machine/machine_rp2_usb.go
+++ b/src/machine/machine_rp2_usb.go
@@ -141,7 +141,7 @@ func setEPDataPID(ep uint32, dataOne bool) {
 }
 
 func SendZlp() {
-	sendUSBPacket(0, empty[:])
+	sendUSBPacket(0, nil)
 }
 
 func sendViaEPIn(ep uint32, data []byte, count int) {

--- a/src/runtime/gc_leaking.go
+++ b/src/runtime/gc_leaking.go
@@ -8,10 +8,15 @@ package runtime
 
 import (
 	"internal/task"
+	"sync/atomic"
 	"unsafe"
 )
 
 const needsStaticHeap = true
+
+var gcScanState atomic.Uint32
+
+func scanCurrentStack() {}
 
 // Ever-incrementing pointer: no memory is freed.
 var heapptr uintptr

--- a/src/runtime/gc_none.go
+++ b/src/runtime/gc_none.go
@@ -7,6 +7,7 @@ package runtime
 // targets that have far too little RAM even for the leaking memory allocator.
 
 import (
+	"sync/atomic"
 	"unsafe"
 )
 
@@ -15,6 +16,9 @@ const needsStaticHeap = false
 var gcTotalAlloc uint64 // for runtime.MemStats
 var gcMallocs uint64
 var gcFrees uint64
+var gcScanState atomic.Uint32
+
+func scanCurrentStack() {}
 
 func alloc(size uintptr, layout unsafe.Pointer) unsafe.Pointer
 

